### PR TITLE
add Ringbuffer interface with GoDocs

### PR DESCRIPTION
**Pull request by furkansenharputlu**
#169

### CODE REVIEW

**Comment by furkansenharputlu** [comment]
verify

**Comment by furkansenharputlu** [comment]
verify

File: core/ringbuffer.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	ReadOne(sequence int64) (item interface{}, err error)
+
+	// ReadManyAsync reads a batch of items from this ringbuffer.
+	// If the number of available items starting at sequence is smaller than maxCount,
+	// then this method will not wait for more items to arrive.
+	// Instead, available items will be returned.
+	//
+	// If there are less items available than minCount, then this call will not return a response until
+	// a necessary number of items becomes available.
+	ReadManyAsync(startSequence int64, minCount int32, maxCount int32) (items []interface{}, err error)
```
Comment (sancar) [inline core/ringbuffer.go]: This api does not look like async.  `readMany` is more suited name

File: core/ringbuffer.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	// Using this policy the oldest item is overwritten no matter it is not old enough to retire. Using this policy you are
+	// sacrificing the time-to-live in favor of being able to write.
+	//
+	// Example: if there is a time-to-live of 30 seconds, the buffer is full and the oldest item in the ring has been placed a
+	// second ago, then there are 29 seconds remaining for that item. Using this policy you are going to overwrite no matter
+	// what.
+	OverflowPolicyOverwrite OverflowPolicy = 0
+
+	// Using this policy the call will fail immediately and the oldest item will not be overwritten before it is old enough
+	// to retire. So this policy sacrificing the ability to write in favor of time-to-live.
```
Comment (Serdaro) [inline core/ringbuffer.go]: ...this policy sacrifices the ability...

**Pull request merged**